backend/app/verification.py

```diff
@@ -321,41 +318,60 @@ def _verify_app(appid: str, login, verified: bool):
     elif provider_name == "GnomeGitLab":
         account = models.GnomeAccount.by_user(sqldb, login["user"])
 
-    if account is not None and account.login == username:
-        if verified:
-            verification = models.UserVerifiedApp(
-                app_id=appid, account=login["user"].id, created=func.now()
-            )
-            try:
-                sqldb.session.add(verification)
-                sqldb.session.commit()
-                return JSONResponse(
-                    {
-                        "verified": True,
-                        "method": "login_provider",
-                        "login_provider": provider_name,
-                        "login_name": account.login,
-                    }
-                )
-            except IntegrityError:
-                raise HTTPException(
-                    status_code=400, detail=ErrorDetail.APP_ALREADY_VERIFIED
-                )
-        else:
-            sqldb.session.execute(
-                delete(models.UserVerifiedApp).where(
-                    models.UserVerifiedApp.app_id == appid
-                )
-            )
+    if account is None or account.login != username:
+        raise HTTPException(status_code=403, detail=ErrorDetail.USERNAME_DOES_NOT_MATCH)
+    else:
+        return (provider_name, username)
+
+
+def _verify_app(appid: str, login, verified: bool):
+    """Marks a flatpak as verified. This requires the logged in user to have an account with the correct provider and
+    with the correct username, as specified by the app ID.
+
+    Returns: Same as get_verification_status()."""
+
+    if not login["state"].logged_in():
+        raise HTTPException(status_code=403, detail="not_logged_in")
```

Comment:
It is sometimes 403 and sometimes 401, I propose 401 for not_logged_in detail everywhere (for coherenece)
```suggestion
        raise HTTPException(status_code=401, detail="not_logged_in")
```

Reply:
Sounds good, probably best left to another PR though.